src/main/java/com/minecolonies/api/compatibility/CompatibilityManager.java

```diff
@@ -324,25 +322,23 @@ private static void serializeRegistryIds(
         return buf.readList(b -> b.readResourceLocation());
     }
 
+    private static final StreamCodec<RegistryFriendlyByteBuf, List<RecipeHolder<?>>> RECIPE_LIST_STREAM_CODEC =
+            RecipeHolder.STREAM_CODEC.apply(ByteBufCodecs.list());
+
     private static void serializeCompostRecipes(
       @NotNull final RegistryFriendlyByteBuf buf,
       @NotNull final Map<Item, RecipeHolder<CompostRecipe>> compostRecipes)
     {
-        final CompostRecipe.Serializer serializer = ModRecipeSerializer.CompostRecipeSerializer.get();
         final List<RecipeHolder<CompostRecipe>> recipes = compostRecipes.values().stream().distinct().toList();
-        buf.writeCollection(recipes, (b, holder) -> {
-            b.writeResourceLocation(holder.id());
-            serializer.toNetwork(buf, holder.value());
-        });
+        //RECIPE_LIST_STREAM_CODEC.encode(buf, recipes);
+        buf.writeCollection(recipes, (b, holder) -> RecipeHolder.STREAM_CODEC.encode((RegistryFriendlyByteBuf) b, holder));
```

Comment:
In theory the `RECIPE_LIST_STREAM_CODEC` should be usable here, but I couldn't work out the specific generics-fu to not get it to complain about differences in wildcards.